operator/main.py

```diff
@@ -464,22 +466,203 @@ def handle_modified(core_v1_client, obj):
+        template_args["host_brick_path"] = storage.get("path", "")
+        template_args["kube_hostname"] = storage.get("node", "")
+        # TODO: Understand the need, and usage of suffix
+        template_args["serverpod_name"] = get_brick_hostname(
+            volname,
+            idx,
+            suffix=False
+        )
+        template_args["brick_path"] = "/bricks/%s/data/brick" % volname
+        template_args["brick_index"] = idx
+        template_args["brick_device"] = storage.get("device", "")
+        template_args["pvc_name"] = storage.get("pvc", "")
+        template_args["brick_device_dir"] = get_brick_device_dir(storage)
+        template_args["brick_node_id"] = storage["node_id"]
+        template_args["k8s_dist"] = K8S_DIST
+
+        filename = os.path.join(MANIFESTS_DIR, "server.yaml")
+        template(filename, **template_args)
+        lib_execute(KUBECTL_CMD, DELETE_CMD, "-f", filename)
+        logging.info(logf("Deleted Server pod",
+                          volname=volname,
+                          manifest=filename,
+                          node=storage.get("node", "")))
+
+def get_storages():
+    """Parse list of Storages from JSON input"""
+
+    storage = storage_list.Storage()
+    cmd = [
+        "kubectl", "get", "configmap",
+        "kadalu-info", "-nkadalu", "-ojson"
+    ]
+    try:
+        resp = utils.execute(cmd)
+        #list_storages does not make use of use 'args'
+        storages = storage_list.list_storages(resp.stdout, None)
+    except utils.CommandError as err:
+        utils.command_error(cmd1, err.stderr)
+    except FileNotFoundError:
+        utils.kubectl_cmd_help(args.kubectl_cmd)
+
+    return storages
+
+def set_num_pvs(storages, volname):
+    """Set PV count for all listed storages"""
+
+    for storage in storages:
 
-def handle_deleted():
+        if storage.storage_name != volname:
+            continue
+
+        dbpath = "/bricks/" + storage.storage_name + "/data/brick/stat.db"
+        query = ("select count(pvname) from pv_stats;")
+        cmd = ["kubectl", "exec", "-i",
+                storage.storage_units[0].podname,
+                "-c", "glusterfsd", "-nkadalu", "--", "sqlite3",
+                dbpath,
+                query
+        ]
+
+        try:
+            resp = utils.execute(cmd)
+            parts = resp.stdout.strip().split("\n")
+            storage.total_size_bytes = float(parts[0].strip())
+            pv_stats_parts = parts[1].strip().split("|")
+            storage.pv_count = int(pv_stats_parts[0])
+        except utils.CommandError as err:
+            print("Failed to get size details of the "
+                  "storage \"%s\"" % storage.storage_name,
+                  file=sys.stderr)
+            print(err, file=sys.stderr)
+            print()
+        except FileNotFoundError:
+            # remove 'args' : args not defined
+            utils.kubectl_cmd_help(args.kubectl_cmd)
```

Comment:
Change this to `log`. This exception will not happen inside Operator container since `kubectl` binary will be available always.